source/family/cypress/PSoC6xxx/PSOC6xxx.c

```diff
@@ -0,0 +1,1449 @@
+*  data_32   : pionter for 4-byte API return data
+*  dev_family   : device family ID 0 - CY8C6XX6,CY8C6XX7, 1 - CY8C6XX8,CY8C6XXA
+*
+* Return:
+*  SUCCESS - SROM executed the task successfully
+*  FAILURE - SROM task is not executed successfully and a timeout error occured.
+*            The failure code is stored in the statusCode global variable.
+*
+* Note:
+*  This function is called to perform volatile memory operations like Read,
+*  Write of Flash.
+*
+******************************************************************************/
+static uint8_t PSoC6_CallSromApi(uint32_t params_32, uint32_t *data_32, uint8_t dev_family)
+{
+    uint8_t status;
+    uint8_t apiStatus = 0;
+
+    //  Acquire IPC_STRUCT[2] for DAP
+    status = PSoC6_Ipc_Acquire(IPC_2, IPC_STRUCT_ACQUIRE_TIMEOUT, dev_family);
+
+    if(status == SWD_ACK)
+    {
+        // Make System Call
+    	  // Write to IPC_STRUCT2.DATA - Sys call ID and Parameters OR address in SRAM, where they are located
+        if ((params_32 & MXS40_SROMAPI_DATA_LOCATION_MSK) == 0)
+        {
+					  if (dev_family == PSOC6_67)
+						{
+							 status = SWD_Send(IPC_STRUCT_DATA_1M, SRAM_SCRATCH_ADDR);
+						}
+						else
+						{
+							 status = SWD_Send(IPC_STRUCT_DATA_2M, SRAM_SCRATCH_ADDR);
+						}
+        }
+        else
+        {
+					 if (dev_family == PSOC6_67)
+					 {
+							status = SWD_Send(IPC_STRUCT_DATA_1M, params_32);
+					 }
+					 else
+					 {
+							status = SWD_Send(IPC_STRUCT_DATA_2M, params_32);
+					 }
+        }
+
+        if ( status == SWD_ACK)
+        {
+            // Enable notification interrupt of IPC_INTR_STRUCT0(CM0+) for IPC_STRUCT2
+					  if (dev_family == PSOC6_67)
+					  {
+								status = SWD_Send(IPC_INTR_STRUCT_DATA_1M, IPC_INTR_STRUCT_DATA_VALUE);
+						}
+						else
+						{
+								status = SWD_Send(IPC_INTR_STRUCT_DATA_2M, IPC_INTR_STRUCT_DATA_VALUE);
+						}
+
+            if ( status == SWD_ACK)
+            {
+                // Notify to IPC_INTR_STRUCT0. IPC_STRUCT2.MASK <- Notify
+							 if (dev_family == PSOC6_67)
+					     {
+									status = SWD_Send(IPC_STRUCT2_NOTIFY_DATA_1M, 0x00000001u);
+							 }
+							 else
+							 {
+									status = SWD_Send(IPC_STRUCT2_NOTIFY_DATA_2M, 0x00000001u);
+							 }
+
+               if ( status == SWD_ACK)
+               {
+                   // Poll lock status
+                   status = PSoC6_Ipc_PollLockStatus(2u ,0u , IPC_STRUCT_ACQUIRE_TIMEOUT, dev_family);
+
+                   if(status == SWD_ACK)
+                   {
+                       // Poll Data byte
+                       if ((params_32 & MXS40_SROMAPI_DATA_LOCATION_MSK) == 0)
+                       {
+                           status = PSoC6_PollSromApiStatus(SRAM_SCRATCH_ADDR, IPC_STRUCT_DATA_TIMEOUT, data_32);
+                       }
+                       else
+                       {
+												  if (dev_family == PSOC6_67)
```

Comment:
Just a reminder to fix indentation on this file.

Reply:
Indentation has been fixed